ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.ClusterAgentsFactory;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.Config;
+import com.github.ambry.config.Default;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.tools.util.ToolUtils;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Reformats all the stores on a given disk.
+ *
+ * Guide to using this tool:
+ * 1. Make sure that the Ambry server is completely shut down before using this tool (i.e there should be no else using
+ * those directories).
+ * 2. Make sure that the scratch space provided has enough space for the largest partition on the disk. If the same
+ * scratch space is used for multiple disks, ensure that it has enough space for all.
+ *
+ * On failures:
+ * The tool does not lose any data but may leave things in an inconsistent state after failures.
```

Comment:
Going forward, I think this can be automated by a "recovery" component in this tool (a bit out of scope for this patch). That would require that the store's actions are logged in a structured way (something that the tool can read and figure out the state from). A bit like Write Ahead Logging.